### PR TITLE
Fix conversion error

### PR DESCRIPTION
**Pull request by sumedhe**
## Purpose
Fix conversion error.

Changes:
* Fix the conversion error by changing the http body as a `json` instead of `string`.
* Fix the issue with getting the status code by parsing it to a `string`.

Fixes: https://github.com/wso2-enterprise/asgardeo-product/issues/18826

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: scim/utils.bal [hunk 1]
```diff
@@ -27,12 +27,12 @@ isolated function getparams(string[] array) returns string {
 
 isolated function getErrorResponse(http:ClientError clientError) returns ErrorResponse|error {
     http:Detail errorDetail = check clientError.detail().ensureType();
-    string responseBodyString = check errorDetail.body.cloneWithType();
-    json responseBody = check responseBodyString.fromJsonString();
+    json responseBody = check errorDetail.body.ensureType();
```
Comment (ThisaruGuruge) [inline scim/utils.bal:30]: Better to use a specific type here, instead of json.

File: scim/utils.bal [hunk 2]
```diff
@@ -27,12 +27,12 @@ isolated function getparams(string[] array) returns string {
...
     string[] schemas = check (check responseBody.schemas).cloneWithType();
+
     return error ErrorResponse (
         string `SCIM Error ${clientError.message()}`, 
         detail = check responseBody.detail, 
-        status = check responseBody.status, 
+        status = (check responseBody.status).toString(),
```
Comment (ThisaruGuruge) [inline scim/utils.bal:36]: Returning these errors is not optimal IMO.

File: scim/utils.bal [hunk 3]
```diff
@@ -27,12 +27,12 @@ isolated function getparams(string[] array) returns string {
...
-    json responseBody = check responseBodyString.fromJsonString();
+    json responseBody = check errorDetail.body.ensureType();
     string[] schemas = check (check responseBody.schemas).cloneWithType();
+
     return error ErrorResponse (
         string `SCIM Error ${clientError.message()}`, 
         detail = check responseBody.detail, 
-        status = check responseBody.status, 
+        status = (check responseBody.status).toString(), 
         schemas = schemas);
```
Comment (ThisaruGuruge) [inline scim/utils.bal:37]: Check the formatting here.

**Comment by NipunaRanasinghe** [comment]
Closing as outdated. @sumedhe please feel free to reopen after updating the PR, if this is still relevant

**Pull request closed without merging**